TechnicalDocuments/Proposals/BuildersToFunctions.md

```diff
@@ -0,0 +1,161 @@
+                         addressSwitches: builders.addressSwitches)
+```
+
+With the free functions approach, this would never be possible to do, since builders don't hold state:
+
+```swift
+extension Builder {
+    enum ClinicalRecords {
+        func makeAppointmentList(session: UserSession,
+                                 enableClinicalDownloads: Bool,
+                                 addressSwitches: AddressSwitchesProtocol,
+                                 primary: Flow,
+                                 secondary: Flow) -> UIViewController
+    }
+}
+```
+
+At the call site, we would replace this:
+
+```swift
+case let .appointmentList(session):
+    builders
+            .appointmentList(with: session,
+                             enableClinicalDownloads: builders.enableClinicalDownloads,
+                             addressSwitches: builders.addressSwitches)
+            .make(primary: navigation, secondary: modal)
+            |> navigation.present
+```
+
+With this:
+
+```swift
+case let .appointmentList(session):
+    Builder.ClinicalRecords
+            .makeAppointmentList(with: session,
+                                enableClinicalDownloads: enableClinicalDownloads,
+                                addressSwitches: addressSwitches,
+                                primary: navigation,
+                                secondary: modal)
+            |> navigation.present
+```
+
+## Impact on an existing codebase
+
+As Ilya pointed out it would make unit testing a bit more cumbersome. That's the case because you no longer inject builders. I would like to make the case for the following:
+
+1. We don't test `FlowControllers` with Unit tests anyway. This is not a reason to not do it, it's simply difficult to justify time doing so, when there are more important pieces to be tested and those tests would bring, in my opinion, little value. More value would be achieved by the next point (**2.**).
+2. We can test that a particular navigation/flow is done as expected independently of this proposal being approved or not. For that to happen we simply observe a ViewModel state and assert on it, since we derive routes from the state.
+3. If we "really" want to test FlowControllers, it's still possible by mocking `Current`, which is something trivial to achieve (`World.init()`). The difference here is that we can't say we are **Unit** testing FlowControllers, but instead these become **Integration** tests. I personally don't feel this is semantically significant. 
+
+Given all those points, it has no impact, since we don't test that layer and there are currently no plans in the short term to do so. 
+
+There is still the question about moving `UserSession`, or something in that shape, to `Current`. Based on an initial conversation with Anders, and quoting him, it would be a "fairly low" effort. This is for me a point that should be thought carefully and be a reason to reject this proposal. 
```

Comment:
```suggestion
There is still the question about moving `UserSession`, or something in that shape, to `Current`. Based on an initial conversation with Anders, and quoting him, it would be a "fairly low" effort. This is for me a point that should be thought about carefully and could be a reason to reject this proposal. 
```